File: csharp/07 - Demo 3 - Refactoring/Refactoring.Tests/ReplaceTempWithQuery/ProductPriceTest.cs
using Xunit;
using FluentAssertions;
using Refactoring.ReplaceTempWithQuery;

namespace Refactoring.Tests.ReplaceTempWithQuery
{
    public class ProductPriceTest
    {
        [Fact]
        public void ProductPriceIsQuantityTimesItemPriceReducedWithDiscount()
        {
            CalculateProductPrice(10, 50).Should().Be(490.0);
        }

        [Fact]
        public void ProductPriceGetsHigherDiscountIfBasePriceIsGreaterThen1000()
        {
            CalculateProductPrice(10, 100).Should().Be(980.0);
            CalculateProductPrice(10, 101).Should().Be(959.5);
        }

        private double CalculateProductPrice(int quantity, int itemPrice)
        {
            ProductPrice productPrice = new ProductPrice(quantity, itemPrice);
            return productPrice.Price;
        }
    }
}

File: java/07 - Demo 3 - Refactoring/src/test/java/replacetempwithquery/ProductPriceTest.java
package replacetempwithquery;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class ProductPriceTest {

    @Test
    public void productPriceIsQuantityTimesItemPriceReducedWithDiscount() {
        assertEquals(490.0, calculateProductPrice(10, 50), 0.0);
    }

    @Test
    public void productPriceGetsHigherDiscountIfBasePriceIsGreaterThen1000() {
        assertEquals(980.0, calculateProductPrice(10, 100), 0.0);
        assertEquals(959.5, calculateProductPrice(10, 101), 0.0);
    }

    private double calculateProductPrice(int quantity, int itemPrice) {
        ProductPrice productPrice = new ProductPrice(quantity, itemPrice);
        double result = productPrice.getPrice();
        return result;
    }
}
